### DISCUSSION TITLE
IsVisible=false element gets loaded in element tree

### DISCUSSION THREAD

**Discussion by saterijst1** [dotnet/maui#26089] [Q&A] [answered]
I'm building an app and I'm creating a collapse in a ListView using a Grid and the `IsVisible` property. I'm hiding these Grids by default but are still loaded in the element tree. This element along with it's children are present in the Live Visual Tree explorer. When I select an element in the explorer, the `IsVisible` property is false as you would expect but the `IsLoaded` property is true. These properties are both true when inspecting the children of the Grid.

Am I missing something? Is my understanding of the relation between `IsVisible` and `IsLoaded` wrong? 

On a side note, [the documentation](https://learn.microsoft.com/en-us/dotnet/maui/deployment/performance?view=net-maui-9.0#reduce-the-visual-tree-size) says the following, so maybe that's where my confusion comes from:

> The [IsVisible](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.visualelement.isvisible#microsoft-maui-controls-visualelement-isvisible) property of each element determines whether the element should be part of the visual tree or not.

**Comment by saterijst1** [comment]
I marked this as resolved but since than I have deleted that comment because it was not resolved. Turned out I was watching the wrong element...

**Comment by bhavanesh2001** [comment, marked as answer]
I think the description[ here](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.visualelement.isvisible?view=net-maui-9.0#microsoft-maui-controls-visualelement-isvisible)  is accurate.

> Gets or sets a value that determines whether this element will be visible on screen and take up space in layouts. This is a bindable property.

The description from[ here](https://learn.microsoft.com/en-us/dotnet/maui/deployment/performance?view=net-maui-9.0#reduce-the-visual-tree-size) is nuanced and could be misinterpreted

> The [IsVisible](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.visualelement.isvisible#microsoft-maui-controls-visualelement-isvisible) property of each element determines whether the element should be part of the visual tree or not.

It implies that **IsVisible = false** might remove the element from the visual tree, which is not accurate.

I raised an[ issue](https://github.com/dotnet/docs-maui/issues/2666) for the same.

**Comment by saterijst1** [comment]
@bhavanesh2001 thanks for the links and pointing out the differences. I agree that the documentation in the first link you have provided is accurate. The second link you posted, to which I also refered in my initial post, is misleading in my opinion.

Thank you for creating the issue!
